src/codec.rs

```diff
@@ -207,11 +207,21 @@ impl<T: arrayvec::Array<Item=u8>> Output for ArrayVecWrapper<T> {
 	}
 }
 
+pub struct PrivateOptimisation(());
```

Comment:
Why do we need this?

Reply:
I want to make the Yes variant not instantiable from outside of this crate

Reply:
You don't export the enum, so it should not be instantiable from the outside at all.

Reply:
done